From 9cc91ebfd3512bd5b4fdea16d8c3c2c31e25dad8 Mon Sep 17 00:00:00 2001
From: Ajay Kumar Gupta <ajay.gupta@ti.com>
Date: Sat, 1 Oct 2011 23:14:22 +0530
Subject: [PATCH 22/29] usb: musb_gadget: fix panic with dual gadget rmmod

removed usb_gadget->disconnect call from musb_gadet.c file as now it
will be taken care by udc core.
---
 drivers/usb/musb/musb_gadget.c |    4 ----
 1 files changed, 0 insertions(+), 4 deletions(-)

diff --git a/drivers/usb/musb/musb_gadget.c b/drivers/usb/musb/musb_gadget.c
index f7fba6b..861def8 100644
--- a/drivers/usb/musb/musb_gadget.c
+++ b/drivers/usb/musb/musb_gadget.c
@@ -1998,10 +1998,6 @@ static void stop_activity(struct musb *musb, struct usb_gadget_driver *driver)
 					nuke(&hw_ep->ep_out, -ESHUTDOWN);
 			}
 		}
-
-		spin_unlock(&musb->lock);
-		driver->disconnect(&musb->g);
-		spin_lock(&musb->lock);
 	}
 }
 
-- 
1.6.6.1

